api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticFilters.java

```diff
@@ -0,0 +1,395 @@
+        break;
+      case BETWEEN:
+        rq.gt(left).lt(right);
+        break;
+      case LIKE:
+      case IN:
+      case EQUAL:
+      case NOT_EQUAL:
+      default:
+        throw new BadRequestException("Bad operator for date modifier: " + mod.getOperator());
+    }
+    return rq;
+  }
+
+  private static boolean isStandardConcept(SearchParameter param) {
+    TreeType paramType = TreeType.valueOf(param.getType());
+    return STANDARD_TREES.contains(paramType);
+  }
+
+  private Set<String> toleafConceptIds(List<SearchParameter> params) {
+    Set<String> out = Sets.newHashSet();
+    for (SearchParameter param : params) {
+      if (param.getGroup()) {
+        out.addAll(
+            childrenByCriteriaGroup.get(param)
+                .stream()
+                .map(id -> Long.toString(id))
+                .collect(Collectors.toSet()));
+      } else {
+        out.add(Long.toString(param.getConceptId()));
+      }
+    }
+    return out;
+  }
+
+  private static class FullTreeType {
+    final TreeType type;
+    final TreeSubType subType;
+    private FullTreeType(TreeType type, TreeSubType subType) {
+      this.type = type;
+      this.subType = subType;
+    }
+
+    static FullTreeType fromParam(SearchParameter param) {
+      return new FullTreeType(
+          TreeType.valueOf(param.getType()), TreeSubType.valueOf(param.getSubtype()));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      FullTreeType that = (FullTreeType) o;
+      return type == that.type &&
+          subType == that.subType;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(type, subType);
+    }
+  }
+
+  private Map<SearchParameter, Set<Long>> buildCriteriaGroupLookup(SearchRequest req) {
+    // Three categories of criteria groups are currently supported in a SearchRequest:
+    // 1. Groups with only tree types specified, e.g. PPI.
+    // 2. Groups with tree types & a criteria code specified, e.g. ICD9/ICD10.
+    // 3. Groups with tree types & a concept ID specified, e.g. drugs.
+    Map<FullTreeType, Set<Long>> childrenByTreeType = Maps.newHashMap();
+    Map<FullTreeType, Map<String, Set<Long>>> childrenByParentCode = Maps.newHashMap();
+    Map<FullTreeType, Map<Long, Set<Long>>> childrenByParentConcept = Maps.newHashMap();
+
+    // Within each category/tree type combination, we can batch MySQL requests to translate these
+    // groups into child concept IDs. First we build up maps to denote which groups to query and
+    // which will eventually hold the results (for now, we mark them with an empty set).
+    for (SearchGroup sg : Iterables.concat(req.getIncludes(), req.getExcludes())) {
+      for (SearchGroupItem sgi : sg.getItems()) {
+        for (SearchParameter param : sgi.getSearchParameters()) {
+          if (!param.getGroup()) {
```

Comment:
If a CB search request contains #2 we only want to count participants that have answered anything for that specific question(criteria already has that question concept_id). If criteriaDao#findCriteriaLeavesByType is called in this case it will return all answers for specific survey(type and subtype). This is not intended.